file_formats/linear_format.mli

```diff
@@ -25,7 +25,7 @@ type linear_item_info =
 
 type linear_unit_info =
   {
-    mutable unit_name : string;
+    mutable unit_name : Compilation_unit.t;
```

Comment:
This probably should be `CU.t` rather than `CU.Name.t`, but it seems like this might be a change -- maybe this was just the `CU.Name.t` equivalent previously?

Reply:
It is a change, here and elsewhere. I think at the time I thought it was obvious that we'd want to include the prefix in lots of places we didn't before, but that's less clear to me now.

Reply:
Oh, I just noticed there's also a `for_pack` field that's a `string option`. We _definitely_ want to replace all that mess with a single `CU.t`.

Reply:
I kept it as `CU.t` and removed the now-redundant `for_pack` field. Changed `Cfg_format` similarly.